sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -91,17 +91,18 @@ var amqpConditionsToRecoveryKind = map[amqp.ErrorCondition]recoveryKind{
 	amqp.ErrorCondition("com.microsoft:operation-cancelled"): RecoveryKindNone,
 
 	// Link recovery needed
-	amqp.ErrorDetachForced: RecoveryKindLink, // "amqp:link:detach-forced"
+	amqp.ErrorDetachForced:          RecoveryKindLink, // "amqp:link:detach-forced"
+	amqp.ErrorTransferLimitExceeded: RecoveryKindLink,
```

Comment:
Curious if you've actually seen this error, or just adding it for thoroughness.

Reply:
It was in the original list of AMQP codes handled in JS so I brought it over but I've not seen it in the wild.

I'm not entirely sure if this error can even surface for us, but, from the service, I think it would mean that we've exceeded the transfer credit for the link. At best, I think we can try to "recover" it by terminating the link, which would cause outgoing messages to stop and be retried against a new link (ie, with fresh credits).